### PR TITLE
Add Cap to keeper liquidation fee

### PR DESCRIPTION
**Pull request by leomassazza**
This PR introduces a new parameter `maxKeeperFee` and use it to cap keeper liquidation fee

### CODE REVIEW

File: contracts/MixinPerpsV2MarketSettings.sol [hunk 1]
```diff
@@ -38,6 +38,8 @@ contract MixinPerpsV2MarketSettings is MixinResolver {
     // Global settings
     // minimum liquidation fee payable to liquidator
     bytes32 internal constant SETTING_MIN_KEEPER_FEE = "futuresMinKeeperFee";
+    // maximum liquidation fee payable to liquidator
+    bytes32 internal constant SETTING_MAX_KEEPER_FEE = "futuresMaxKeeperFee";
```
Comment (davidvuong) [inline contracts/MixinPerpsV2MarketSettings.sol:42]: Dang, we forgot to rename `futures` to `perps` here. How difficult would it makes deploys if we were to rename both to be `perps{Min,Max}KeeperFee`?

File: contracts/PerpsV2MarketBase.sol [hunk 2]
```diff
@@ -359,18 +359,21 @@ contract PerpsV2MarketBase is Owned, MixinPerpsV2MarketSettings, IPerpsV2MarketB
...
+     * liquidations of small positions, or overpaying.
      * @param positionSize size of position in fixed point decimal baseAsset units
      * @param price price of single baseAsset unit in sUSD fixed point decimal units
      * @return lFee liquidation fee to be paid to liquidator in sUSD fixed point decimal units
      */
     function _liquidationFee(int positionSize, uint price) internal view returns (uint lFee) {
         // size * price * fee-ratio
         uint proportionalFee = _abs(positionSize).multiplyDecimal(price).multiplyDecimal(_liquidationFeeRatio());
+        uint maxFee = _maxKeeperFee();
+        uint topedProportionalFee = proportionalFee > maxFee ? maxFee : proportionalFee;
```
Comment (davidvuong) [inline contracts/PerpsV2MarketBase.sol:372]: Do you mean `topped` here instead of `toped`?

File: test/contracts/PerpsV2Market.js [hunk 3]
```diff
@@ -5108,6 +5109,42 @@ contract('PerpsV2Market PerpsV2MarketAtomic', accounts => {
 				await perpsV2MarketSettings.setLiquidationFeeRatio(toUnit(0.02), { from: owner });
 				assert.bnEqual(await perpsV2Market.liquidationFee(trader2), toUnit(60));
 			});
+
+			it('top up liquidation fee', async () => {
```
Comment (davidvuong) [inline test/contracts/PerpsV2Market.js:5113]: nit: Maybe rename this test to be 'max liquidation fee' for clarity but test looks good 👍

**Review by davidvuong — APPROVED** [review]
LGTM. Just a typo and some questions.

**Comment by davidvuong** [comment]
Minor: Could you also write a test that checks if the fee pool grows by `remainingMargin - liqFees`? Unless I'm mistaken, looking at the `liquidatePosition`, only the `order.commitFee` is sent to the fee pool

File: contracts/MixinPerpsV2MarketSettings.sol (hunk 1 above)
Comment (leomassazza) [reply contracts/MixinPerpsV2MarketSettings.sol:42]: not that difficult, but we need to be sure we set all params to the right value when deploying the new contract

Comment (davidvuong) [reply contracts/MixinPerpsV2MarketSettings.sol:42]: Cool. Might be worth doing this after this goes out.

**Review by davidvuong — APPROVED** [review]

**Pull request merged**